src/Morpho.sol

```diff
@@ -272,7 +273,8 @@ contract Morpho is IMorpho {
 
         position[id][onBehalf].borrowShares -= shares.toUint128();
         market[id].totalBorrowShares -= shares.toUint128();
-        market[id].totalBorrowAssets = UtilsLib.zeroFloorSub(market[id].totalBorrowAssets, assets).toUint128();
+        // Safe "unchecked" cast since the value is smaller than totalBorrowAssets.
+        market[id].totalBorrowAssets = uint128(UtilsLib.zeroFloorSub(market[id].totalBorrowAssets, assets));
```

Comment:
I don't think that we should add think kind of optimization at this stage (even earlier in the development actually), similarly to #556. We don't know how much gas does it save right? (no veto though, it's correct indeed).

Reply:
I unfortunately agree that this PR has the exact same stakes than #556 so we shouldn't merge it... Too bad we realize late than we've complexified our lives